persist/fs/write.go

```diff
@@ -117,36 +120,71 @@ func NewWriter(opts Options) (FileSetWriter, error) {
 // Open initializes the internal state for writing to the given shard,
 // specifically creating the shard directory if it doesn't exist, and
 // opening / truncating files associated with that shard for writing.
-func (w *writer) Open(
-	namespace ident.ID,
-	blockSize time.Duration,
-	shard uint32,
-	blockStart time.Time,
-) error {
-	shardDir := ShardDirPath(w.filePathPrefix, namespace, shard)
-	if err := os.MkdirAll(shardDir, w.newDirectoryMode); err != nil {
-		return err
-	}
-	w.blockSize = blockSize
-	w.start = blockStart
+func (w *writer) Open(opts WriterOpenOptions) error {
+	var (
+		shardDir              string
+		fileTimestampUnixNano = opts.BlockStart
```

Comment:
super nit: could you just use `opts.BlockStart` instead of this var. it's shorter and more obvious while reading the code below to see the `opts.BlockStart` reference

Reply:
yep done